frame/balances/src/tests.rs

```diff
@@ -683,6 +694,48 @@ macro_rules! decl_tests {
 				});
 		}
 
+		#[test]
+		fn emit_events_with_reserve_and_unreserve() {
+			<$ext_builder>::default()
+				.build()
+				.execute_with(|| {
+					let _ = Balances::deposit_creating(&1, 100);
+
+					System::set_block_number(2);
+					let _ = Balances::reserve(&1, 10);
+
+					assert_eq!(
+						events(),
+						[
+							Event::system(system::RawEvent::NewAccount(1)),
+							Event::balances(RawEvent::Endowed(1, 100)),
```

Comment:
I would suggest modifying the test here to not check for other events that are not within the scope of this test. Will help prevent other future changes from breaking this test.

So i would index into the last element of the vec, and simply check just that one item matches reserved